lib/report.js

```diff
@@ -158,7 +160,162 @@ class ConciseReporter extends Reporter {
   }
 }
 
+class TapReporter extends Reporter {
+
+  constructor(options) {
+    super(options);
+    this.successful = 0;
+    this.failed = 0;
+    this.skipped = 0;
+    this.failedTests = [];
+    this.testNumeration = [1];
+    this.lastIndex = 0;
+    this.message = '';
+    this.firstFailure = true;
+    this.got = [];
+    this.expected = [];
+  }
+
+  error(test, error) {
+    // remove one success because 'error' may only be received after end
+    // though the test may have already failed so we need a way to check
+    // if we should do this, i.e. store map of successful tests
+    // this.successes--;
+    // test.todo ? this.skipped++ : this.failures++;
+    // TODO(lundibundi): add test.metadata.filename here too once available
+    this.listFailure(
+      test,
+      { message: `Failure after end: ${error ? error.message : 'no message'}` },
+      ` Test failed: ${test.caption}`
+    );
+  }
+
+  lastElement(inc) {
+    if (inc) this.testNumeration[this.lastIndex]++;
```

Comment:
It looks like only the last element of `this.testNumeration` is used, maybe it shouldn't be an array but just a value?

Reply:
For example, if we've 3 tests and one subtest with successful asserts. How can I control the number inside and outside(if after the subtest we've another tests) the subtest?